In [6]:
import pandas as pd
import numpy as np

In [8]:
df = pd.read_csv("/content/dataset_4_V.csv")
df.head()

ID     Tenure VisibleMinority  Age  Gender  EvaluationScore
0   1    Tenured              No   55    Male             3.06
1   2    Tenured              No   50    Male             3.66
2   3    Tenured             Yes   64  Female             2.05
3   4  Untenured             Yes   46  Female             1.28
4   5    Tenured             Yes   56  Female             1.03

# Q1. Using the teachers' rating dataset, what is the probability of receiving an evaluation score of greater than 4.5

In [9]:
count_gt_4_5 = (df['EvaluationScore'] > 4.5).mean()
print(f"Probability of receiving an evaluation score greater than 4.5: {count_gt_4_5:.3f}")

Probability of receiving an evaluation score greater than 4.5: 0.100


There is a 12% probability that a professor gets an evaluation score greater than 4.5.

# Q2. Using the teachers' rating dataset, what is the probability of receiving an evaluation score greater than 3.5 and less than 4.2

In [10]:
prob_between = ((df['EvaluationScore'] > 3.5) & (df['EvaluationScore'] < 4.2)).mean()
print(f"Probability of receiving an evaluation score between 3.5 and 4.2: {prob_between:.3f}")

Probability of receiving an evaluation score between 3.5 and 4.2: 0.160


Here, 16% of professors have received an evaluation score between 3.5 and 4.2.

This indicates that moderate-to-high ratings occur occasionally but are not the majority.

# Q3. Using the two-tailed test from a normal distribution:
# 1. A professional basketball team wants to compare its performance with that of players in a regional league.
# 2. The pros are known to have a historic mean of 12 points per game with a standard deviation of 5.5.
# 3. A group of 36 regional players recorded on average 10.7 points per game.

Null Hypothesis (H₀):

The mean points of regional players is not different from the historic mean of 12.

Alternative Hypothesis (H₁):

The mean points of regional players is different from the historic mean of 12.

Calculate Z-score for two-tailed test

Theory:

Z = (sample_mean − population_mean) / (population_std / √n)

Compare Z with critical value from standard normal distribution for α = 0.05 (two-tailed → ±1.96).

In [11]:
mu = 12
sigma = 5.5
n = 36
x_bar = 10.7

z = (x_bar - mu) / (sigma / np.sqrt(n))
z


np.float64(-1.418181818181819)

Critical Z-value for a 95% confidence level (two-tailed test) = ±1.96

Decision rule:

If |Z| > 1.96 → Reject H₀

If |Z| ≤ 1.96 → Fail to reject H₀

Here, |Z| = 1.418 < 1.96 → Fail to reject the null hypothesis

There is no significant evidence to suggest that the mean points of the regional players are different from the historic mean of 12.

The regional players’ average performance (10.7 points per game) is not significantly different from the pros’ historic average.